### PR TITLE
feat: remove `@scalar/themes` from the dependencies of `@scalar/types`

### PR DESCRIPTION
**Pull request by inyourtime**
#3241

### CODE REVIEW

**Review by hanspagel — CHANGES_REQUESTED** [review]
Let’s try to add types as a dependency of themes and import the ThemeId there instead of copying the type ✌️

**Comment by inyourtime** [comment]
i will do it.

File: packages/themes/tsconfig.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
     "baseUrl": ".",
     "target": "ES2022" /* Set the JavaScript language version for emitted JavaScript and include compatible library declarations. */,
     "module": "ESNext" /* Specify what module code is generated. */,
-    "moduleResolution": "Node",
+    "moduleResolution": "Bundler",
```
Comment (amritk) [inline packages/themes/tsconfig.json:6]: hey great work! do you think you can do one better and just extend the base tsconfig?

```json
{
  "extends": "../../tsconfig.json"
}
```

**Review by amritk — CHANGES_REQUESTED** [review]
Just requested one change to the tsconfig, but excellent work!

**Review by amritk — APPROVED** [review]
LGTM!

**Review by hanspagel — APPROVED** [review]
Thanks so much! :) Love when the dependency tree becomes smaller/simpler.

**Pull request merged**